ilastik/config.py

```diff
@@ -44,6 +49,70 @@
 logging_config: ~/custom_ilastik_logging_config.json
 """
 
+
+@dataclass
+class Increment:
+    increment: int
+
+
+class _ConfigBase(BaseModel):
+    model_config = ConfigDict(extra="ignore", use_attribute_docstrings=True)
+
+
+class IlastikSection(_ConfigBase):
+
+    plugin_directories: str = "~/.ilastik/plugins,"
```

Comment:
your comment sent me into a "Sinnkrise" -- the first step to overengineering ;)

Currently validation on all of the strings is very nimble. Especially for the comma-separated list of paths - where `,` is a valid character for Paths on all OSses... So a "strip whitespace" validator will not do the job. This does not change with this PR, but it _does_ make it more accessible and hence more prone to cause problems.

without major work I see 3 options

1) scrap this
2) keep as is, and accept that changing the string values here can lead to all kinds of errors.
3) hide at least some of the options that we don't want to expose to all users

I'm leaning towards 3)... which I implemented in https://github.com/ilastik/ilastik/pull/2899/commits/5bc90554eaf906f1a8d537ead919171432b5d569 (also happy to remove if too controversial)

Reply:
what do you think @btbest? :)

Reply:
Sorry, I got bogged down in details instead of maybe just replying... I also like 3 the most - especially as advanced options that are still available if the user really looks for them, as you've implemented now :)